spec/support/shared_examples/abilities.rb

```diff
@@ -0,0 +1,95 @@
+# collections of shared examples
+
+# logged in, unconfirmed:
+shared_examples "same as guest user" do
+  it_behaves_like "can view public pages"
+  it_behaves_like "can not modify things on public pages"
+  it_behaves_like "can not create new user"
+  it_behaves_like "can not comment"
+  it_behaves_like "has no access to other user's accounts"
+  it_behaves_like "can not read role restricted or owner restricted pages" # now: ApplicationDraft
+end
+
+# confirmed
+shared_examples "same as logged in user" do
+  it_behaves_like "can view public pages"
+  it_behaves_like "can not create new user"
+  it_behaves_like "can not comment"
+  it_behaves_like "has no access to other user's accounts"
+  it_behaves_like "can not read role restricted or owner restricted pages" # now: ApplicationDraft
+  it_behaves_like "can modify own account"
+end
+
+# different roles
+shared_examples "same public features as confirmed user" do
+  it_behaves_like "can view public pages"
+  it_behaves_like "can not create new user"
+  it_behaves_like "can modify own account"
+  # todo can creates comments; needs work
+end
+
+
+# DETAILS, in order of appearance
+
+PUBLIC_INDEX_PAGES = [Activity, User, Team, Project, Conference].freeze
+
+# details for unrestricted
+shared_examples 'can view public pages' do
+  PUBLIC_INDEX_PAGES.each do |page|
+    it { expect(subject).to be_able_to(:read, page) }
```

Comment:
Same here: you can ditch the `subject` when using this syntax:
```ruby
it { is_expected.to be_able_to(:read, page) }
```

Same for the rest of this file of course.

Reply:
I hate that syntax with a passion. 😅 There is no performance or functional difference, right?

Reply:
Hm, I tbh. don't know, but probably not 🤔 

I personally think the syntax reads way more natural than explictly saying "subject". Take for instance
> describe thing - it is expected to be such a thing

Since I haven't yet been on a project where linters would allow you syntax 🚔 🚒  - I'm probably also just used to the implicit one 😂